python/ray/serve/_private/deployment_state.py

```diff
@@ -1227,6 +1227,17 @@ def get_running_replica_infos(self) -> List[RunningReplicaInfo]:
             for replica in self._replicas.get([ReplicaState.RUNNING])
         ]
 
+    def get_running_replica_node_ids(self) -> Set[str]:
+        """Get the node ids of all running replicas in this deployment.
+
+        This is used to determine which node has replicas. Only nodes with replicas and
+        head node should have active proxies.
+        """
+        return {
+            replica.actor_node_id
+            for replica in self._replicas.get([ReplicaState.RUNNING])
```

Comment:
We should expand the states here to include: `{STARTING, UPDATING, RECOVERING, RUNNING}` (because the other 3 will transition to `RUNNING` in the common case, don't want to downscale right when they start receiving traffic)

maybe then rename the method `get_active_node_ids` ?